src/utils/format_stackdriver/format_stackdriver.c

```diff
@@ -393,43 +364,41 @@ static int format_metric(yajl_gen gen, metric_t const *m) {
  * Stackdriver due to lack of state. */
 static int format_time_series(yajl_gen gen, metric_t const *m,
                               sd_resource_t *res) {
-  metric_type_t type = m->family->type;
-
-  cdtime_t start_time = 0;
-  counter_t start_value = 0;
-  int status = read_cumulative_state(m, &start_time, &start_value);
-  if (status != 0) {
-    return status;
+  uc_first_metric_result_t start = uc_first_metric(m);
+  if (start.err) {
+    ERROR("format_stackdriver: uc_first_metric failed: %s",
+          STRERROR(start.err));
+    return start.err;
   }
-  if (start_time == m->time) {
-    /* for cumulative metrics, the interval must not be zero. */
-    return EAGAIN;
-  }
-  if ((type == METRIC_TYPE_GAUGE) || (type == METRIC_TYPE_UNTYPED)) {
-    double d = (double)m->value.gauge;
-    if (isnan(d) || isinf(d)) {
+
+  switch (m->family->type) {
+  case METRIC_TYPE_GAUGE:
+    if (!isfinite(m->value.gauge)) {
       return EAGAIN;
     }
-  }
-  if (type == METRIC_TYPE_COUNTER) {
-    if (m->value.counter < start_value) {
+  case METRIC_TYPE_COUNTER:
+    // for cumulative metrics the interval must not be zero.
+    if (start.time == m->time) {
       return EAGAIN;
     }
+  case METRIC_TYPE_UNTYPED:
+    ERROR("format_stackdriver: Invalid metric type: %d", m->family->type);
+    return EINVAL;
```

Comment:
I'm working towards treating `METRIC_TYPE_UNTYPED` as an invalid state. Some of this leaked into here, but most of it is in  #4266.

The behavior in `format_typed_value()` was an oversight. Fixed.

Reply:
`format_metric_kind()` and `format_value_type()` still accept METRIC_TYPE_UNTYPED as alias for gauge?

Reply:
That is fixed in #4266.

Reply:
Another inconsistency is one of the 4 switch-cases including `METRIC_TYPE_UNTYPED` is missing `default:`, other 3 do have that.

Reply:
> That is fixed in https://github.com/collectd/collectd/pull/4266

Ok, I'll approve that, the other inconsistencies can be handled there.

Reply:
I have tried hard to avoid having a `default:` case, becase without a default case compilers can (and will) warn you if you don't cover all values of the enum. That is really helpful in finding locations that differentiate by metric type. With the next PR this will be more consistent.